x-pack/legacy/plugins/apm/public/components/app/ServiceNodeMetrics/index.tsx

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import {
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiTitle,
+  EuiHorizontalRule,
+  EuiFlexGrid,
+  EuiPanel,
+  EuiSpacer,
+  EuiStat,
+  EuiToolTip
+} from '@elastic/eui';
+import React from 'react';
+import { i18n } from '@kbn/i18n';
+import styled from 'styled-components';
+import { ApmHeader } from '../../shared/ApmHeader';
+import { useUrlParams } from '../../../hooks/useUrlParams';
+import { useAgentName } from '../../../hooks/useAgentName';
+import { useServiceMetricCharts } from '../../../hooks/useServiceMetricCharts';
+import { ChartsSyncContextProvider } from '../../../context/ChartsSyncContext';
+import { MetricsChart } from '../../shared/charts/MetricsChart';
+import { useFetcher, FETCH_STATUS } from '../../../hooks/useFetcher';
+import { callApmApi } from '../../../services/rest/callApmApi';
+import { truncate, px, unit } from '../../../style/variables';
+
+const INITIAL_DATA = {
+  host: '',
+  containerId: ''
+};
+
+const Truncate = styled.span`
+  display: block;
+  ${truncate(px(unit * 12))}
+`;
```

Comment:
Almost everywhere we use `truncate()` we also use `EuiTooltip` (I found ~8 instances of this). Would it make sense to make a `TruncatedToolTip` component that in addition to `EuiTooltip`'s props also takes a width:
```tsx
<TruncatedToolTip content={host} truncate-width={px(unit * 12)}>{host}</TruncatedToolTip>
```

Reply:
I'll have a look, will need to check if other styling is consistent enough to be abstracted away.

Reply:
Or perhaps this should be a feature request for EUI.

Reply:
I think it's worth it as I just had to create another instance of truncate + tooltip. Created an issue for it: https://github.com/elastic/kibana/issues/47195